backend/dataall/base/utils/expiration_util.py

```diff
@@ -13,7 +13,12 @@ def calculate_expiry_date(expirationPeriod, expirySetting):
             day = calendar.monthrange(quarterlyCalculatedDate.year, quarterlyCalculatedDate.month)[1]
             shareExpiryDate = datetime(quarterlyCalculatedDate.year, quarterlyCalculatedDate.month, day)
         elif expirySetting == Expiration.Monthly.value:
-            monthlyCalculatedDate = currentDate + relativedelta(months=expirationPeriod - 1)
+            if currentDate < datetime(currentDate.year, currentDate.month, 15):
```

Comment:
@dlpzx , can you implement the same for quarterly share(s) ? `expirySetting == Expiration.Quartely.value`. 
This will create uniformity for both these types